src/nplinker/genomics/gcf.py

```diff
@@ -61,10 +60,17 @@ def add_bgc(self, bgc: BGC) -> None:
         """Add a BGC object to the GCF."""
         self._bgcs.add(bgc)
         bgc.add_parent(self)
-        if bgc.strain is not None:
-            self.strains.add(bgc.strain)
-        else:
-            logger.warning("No strain specified for the BGC %s", bgc.bgc_id)
+
+    @property
+    def strains(self) -> StrainCollection:
+        """Get the collection of strains"""
+        sc = StrainCollection()
```

Comment:
Why do we want to redefine this instance every time we call `strains` method?

Reply:
GCF is a group of BGCs, and every BGC has a strain value (`BGC.strain`). So `GCF.strains` is just the collection of strains of its BGC members and should be extracted from `BGC.strain` every time. Creating a new instance make it easier to update the BGC members of BGC without extra work on update strains.